content/articles/getting-started-with-optimization-in-matlab/index.md

```diff
@@ -0,0 +1,196 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /getting-started-with-optimization-in-matlab/
+title: Getting Started with Optimization in MATLAB
+description: In this article we discuss the fundamentals of optimization and cover real world examples where optimization is applied using MATLAB. 
+author: florence-atieno
+date: 2021-08-20T00:00:00-18:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  -url: /engineering-education/getting-started-with-optimization-in-matlab/hero.jpg
+   alt: MATLAB optimization example image
+---
+
+
+Optimization combines the inputs to achieve the optimum output subject to satisfying certain specified constraints and conditions. It involves choosing the best among given data or combining the inputs to obtain the optimal working condition. It is a mathematical discipline that works to get the maxima and minima. Thus, optimization forms a part of linear programming. 
+
+
+### Prerequisites
+- [MATLAB](https://www.mathworks.com/products/get-matlab.html?s_tid=gn_getml) installed.
+- Proper understanding of [MATLAB](https://www.section.io/engineering-education/getting-started-with-matlab/) basics.
+
+### Introduction
+
+Optimization is applicable in a wide range of fields. It includes the financial sector, transport sector, and many more. Hence, it is an essential tool used in everyday life with it's impact felt by billions of people on a daily basis. This article will look at the basics of optimization and how to apply them in Matlab. We will also look at its application in various fields.
+
+An optimization problem is defined by `df(x)/dx=0`. Finding the minimum point in a curve can be done using a while loop. For example, given a function $y(x)= 2x^2+20x-22$, we find its minimum point using a simple while loop as shown below;
+
+```matlab
+clear
+clc
+x = -20: 0.1: 20;
+y = 2.*x.^2 + 20.*x - 22;
+plot(x,y)
+
+grid
+```
+
+Here we plot the function using the `plot` function as shown above. Then, we use the while loop to iterate over all the data points in the curve. When the current value is larger than the previous value, we identify the point as the minimum of the curve. Once done, we print our output using the `fprintf` function. The command window displays the coordinates of the minimum point.
```

Comment:
Check the changes made to this paragraph. You tend to use a lot of we can and then we. Avoid making it repetitive. See the changes made in this paragraph and make changes accordingly.